app/code/Magento/QuoteGraphQl/Model/Resolver/MergeCarts.php

```diff
@@ -31,44 +37,93 @@ class MergeCarts implements ResolverInterface
      */
     private $cartRepository;
 
+    /**
+     * @var CustomerCartResolver
+     */
+    private $customerCartResolver;
+
+    /**
+     * @var QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedQuoteId;
+
     /**
      * @param GetCartForUser $getCartForUser
      * @param CartRepositoryInterface $cartRepository
+     * @param CustomerCartResolver $customerCartResolver
+     * @param QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
      */
     public function __construct(
         GetCartForUser $getCartForUser,
-        CartRepositoryInterface $cartRepository
+        CartRepositoryInterface $cartRepository,
+        CustomerCartResolver $customerCartResolver,
+        QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
     ) {
         $this->getCartForUser = $getCartForUser;
         $this->cartRepository = $cartRepository;
+        $this->customerCartResolver = $customerCartResolver;
+        $this->quoteIdToMaskedQuoteId = $quoteIdToMaskedQuoteId;
     }
 
     /**
      * @inheritdoc
      */
-    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
-    {
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
         if (empty($args['source_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "source_cart_id" is missing'));
-        }
-
-        if (empty($args['destination_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "destination_cart_id" is missing'));
+            throw new GraphQlInputException(__(
+                'Required parameter "source_cart_id" is missing'
```

Comment:
```suggestion
                'Required parameter "destination_cart_id" is missing'
```

Reply:
Hello @rogyar 
Thank you for your review.
This block of the code is needed to check the parameter `source_cart_id`
The function `empty` is suitable for this checking.
Please take a look at the result code by [this link](https://github.com/andrewbess/magento2/blob/CE-30469/app/code/Magento/QuoteGraphQl/Model/Resolver/MergeCarts.php#L78-L82).

Reply:
But you are checking for `destination_cart_id` above, or I miss something? 

```php
if (empty($args['destination_cart_id'])) {
```

Reply:
Sorry, I got your point. Looked at the wrong line of code ;)